dev-tool/src/list_net/data_source.rs

```diff
@@ -309,8 +382,36 @@ impl InMemorySamples {
         Ok(self_)
     }
 
+    /// Returns the number of samples in the storages.
+    ///
+    /// The samples can be referred to by usize ids from
+    /// `0` to `number_of_samples`.
```

Comment:
referred by their id?

Reply:
It's not really their id i.e. the id is not a property of the sample.

It's just a mechanism to temporary refer to a sample without having the sample yet.